src/router/routes/cube/blog.ts

```diff
@@ -156,10 +199,11 @@ export const deleteBlogHandler = async (req: Request, res: Response) => {
     const { id } = req.params;
     const { user } = req;
 
+    //The ensureAuth middleware means this should never occur
     if (!user) {
       req.flash('danger', 'Please login to delete a blog post.');
 
-      return redirect(req, res, `/cube/blog/${encodeURIComponent(id)}`);
+      return redirect(req, res, `/cube/blog/blogpost/${encodeURIComponent(id)}`);
```

Comment:
The redirect wasn't logical before. The Id is the blog post id, but was trying to redirect as if it is a cube id.
That said, since the ensureAuth middleware will redirect the user to login before the handler is called this part of the code is not used.